src/Lock.php

```diff
@@ -50,18 +52,20 @@ public function getId(): int
 
     /**
      * Releases the lock. No-op if the lock has already been released.
+     *
+     * @return Promise<void>
      */
-    public function release()
+    public function release(): Promise
     {
         if (!$this->releaser) {
-            return;
+            return new Success();
```

Comment:
If the previous release failed, this should probably not return `new Success;`. This is the kind of problems you run into once you return a promise. :P

Reply:
@enumag The promise from `$this->releaser` can be stored and returned here instead. We might want to remove passing `$this` to the function to avoid a circular reference.